### PR TITLE
bump openshift-client to 5.7.x

### LINKED ISSUE #427: Update openshift client to latest 5.x
There are some breaking changes, but it shouldn't be complicated to deal with them. See [CHANGELOG.md](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) for details.

### PR DESCRIPTION
**Pull request by simkam**
Although there are no API changes in XTF itself, it brings new kubernetes-client which contains many breaking changes. 

fixes #427 

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

### CODE REVIEW

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java [hunk 1]
```diff
@@ -801,6 +818,26 @@ public Build startBinaryBuild(String buildConfigName, File file) {
...
+    //    at okhttp3.OkHttpClient$Builder.readTimeout(OkHttpClient.java:596)
+    //    at io.fabric8.openshift.client.dsl.internal.build.BuildConfigOperationsImpl.submitToApiServerWithRequestBody(BuildConfigOperationsImpl.java:283)
+    //    at io.fabric8.openshift.client.dsl.internal.build.BuildConfigOperationsImpl.fromInputStream(BuildConfigOperationsImpl.java:186)
+    //    at io.fabric8.openshift.client.dsl.internal.build.BuildConfigOperationsImpl.fromInputStream(BuildConfigOperationsImpl.java:167)
+    //    at io.fabric8.openshift.client.dsl.internal.build.BuildConfigOperationsImpl.fromInputStream(BuildConfigOperationsImpl.java:72)
+    //    at cz.xtf.core.bm.BinaryBuildFromSources.build(BinaryBuildFromSources.java:60)
+    //    at cz.xtf.core.bm.BuildManager.deploy(BuildManager.java:62)
+    //    at cz.xtf.junit5.listeners.ManagedBuildPrebuilder.testPlanExecutionStarted(ManagedBuildPrebuilder.java:85)
+    //
+    // kubernetes-client issue will be created
```
Comment (mnovak1) [inline core/src/main/java/cz/xtf/core/openshift/OpenShift.java:834]: Thanks, could you link the kubernetes-client issue here, please? We might need new XTF issue to remove it once fixed in kubernetes-client so we don't forget it.

**Review by mnovak1 — APPROVED** [review]

**Comment by simkam** [comment]
rebased on master and updated openshift-client to 5.7.3. We are still looking into some issues with this upgrade.

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 1 above)
Comment (simkam) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java:834]: I need to investigate it more, the code is unchanged. It is not obvious to me now why we need this. I'll create XTF issue to evaluate and investigate the workaround.

**Comment by simkam** [comment]
rebased on master and undrafted. It is ready for review. Tests using this upgrade are passing. 
ocp-client-bump
Failures are either unrelated to this change or intermittent.

File: core/src/main/java/cz/xtf/core/openshift/OpenShift.java (hunk 1 above)
Comment (simkam) [reply core/src/main/java/cz/xtf/core/openshift/OpenShift.java:834]: https://github.com/xtf-cz/xtf/issues/461

**Review by mnovak1 — APPROVED** [review]

**Pull request merged**